### PR TITLE
 build: update angular package to `12.0.0-rc.0`

### PR DESCRIPTION
**Pull request by alan-agius4**
Updates angular packages to 12.0.0-rc.0

Closes #41758

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]

**Comment by alan-agius4** [comment]
@petebacondarwin & @AndrewKushnir can you please review for size-tracking? Thanks

File: goldens/size-tracking/aio-payloads.json [hunk 1]
```diff
@@ -2,17 +2,17 @@
...
     }
   },
   "aio-local": {
     "master": {
       "uncompressed": {
-        "runtime-es2015": 3033,
-        "main-es2015": 453725,
+        "runtime-es2015": 3757,
+        "main-es2015": 450346,
         "polyfills-es2015": 55230
```
Comment (petebacondarwin) [inline goldens/size-tracking/aio-payloads.json:16]: This appears to have increase slightly on master already, so we might as well update it here now: `55283`.

(The non-local one too)

File: goldens/size-tracking/aio-payloads.json [hunk 2]
```diff
@@ -2,17 +2,17 @@
...
       }
     }
   },
   "aio-local": {
     "master": {
       "uncompressed": {
-        "runtime-es2015": 3033,
-        "main-es2015": 453725,
+        "runtime-es2015": 3757,
+        "main-es2015": 450346,
```
Comment (petebacondarwin) [inline goldens/size-tracking/aio-payloads.json:15]: Nice - a small decrease!

File: goldens/size-tracking/aio-payloads.json [hunk 3]
```diff
@@ -2,17 +2,17 @@
...
         "polyfills-es2015": 55230
       }
     }
   },
   "aio-local": {
     "master": {
       "uncompressed": {
-        "runtime-es2015": 3033,
-        "main-es2015": 453725,
+        "runtime-es2015": 3757,
```
Comment (petebacondarwin) [inline goldens/size-tracking/aio-payloads.json:14]: Any idea what causes this increase? Is it webpack 5?

**Review by petebacondarwin — APPROVED** [review]
Reviewed-for: size-tracking

File: goldens/size-tracking/aio-payloads.json (hunk 3 above)
Comment (alan-agius4) [reply goldens/size-tracking/aio-payloads.json:14]: Yeah, it is caused by Webpack 5.

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: size-tracking

File: aio/src/app/search/search.service.ts [hunk 4]
```diff
@@ -27,7 +27,8 @@ export class SearchService {
       .pipe(
         concatMap(() => {
           // Create the worker and load the index
-          const worker = new Worker('./search.worker', { type: 'module' });
+          // tslint:disable-next-line: whitespace
+          const worker = new Worker(new URL('./search.worker', import.meta.url), { type: 'module' });
```
Comment (gkalpak) [inline aio/src/app/search/search.service.ts:31]: OOC, why is this needed with v12?

File: aio/angular.json [hunk 5]
```diff
@@ -31,15 +31,10 @@
...
             "webWorkerTsConfig": "tsconfig.worker.json",
-            "aot": true,
-            "optimization": true,
-            "buildOptimizer": true,
             "outputHashing": "all",
             "sourceMap": true,
             "statsJson": true,
             "extractLicenses": true,
-            "namedChunks": true,
-            "vendorChunk": false,
```
Comment (gkalpak) [inline aio/angular.json:42]: Are these the default now?

File: aio/angular.json [hunk 6]
```diff
@@ -33,7 +33,6 @@
             "webWorkerTsConfig": "tsconfig.worker.json",
             "outputHashing": "all",
             "sourceMap": true,
-            "statsJson": true,
```
Comment (gkalpak) [inline aio/angular.json:39]: IIRC we had discussed this in the past and @IgorMinar was keen on keeping the stats.
I might remember incorrectly, but it would be good to check with @IgorMinar (just in case).

File: scripts/puppeteer-chromedriver-versions.js [hunk 7]
```diff
@@ -13,8 +13,8 @@
 // "chromium_revision": "722234" but this does not map easily to the Chrome version
 // so we use this mapping here instead.
 module.exports = {
-  '9.0.0': '90.0.4430.24',
-  '8.0.0': '90.0.4430.24',
+  '9.0.0': '91.0.4472.19',
+  '8.0.0': '91.0.4472.19',
```
Comment (gkalpak) [inline scripts/puppeteer-chromedriver-versions.js:17]: According to [puppeteer/.../main/docs/api.md][1], this should be v90 😕
BTW, we don't need to have all the values. Just the ones we actually use.

[1]: https://github.com/puppeteer/puppeteer/blob/main/docs/api.md

**Review by gkalpak — APPROVED** [review]
A couple of minor comments/questions. Otherwise lgtm :+1: 🚀 :tada:

File: aio/src/app/search/search.service.ts (hunk 4 above)
Comment (alan-agius4) [reply aio/src/app/search/search.service.ts:31]: This is the new syntax needed for Webpack 5.

File: aio/angular.json (hunk 5 above)
Comment (alan-agius4) [reply aio/angular.json:42]: Yes

File: aio/angular.json (hunk 6 above)
Comment (alan-agius4) [reply aio/angular.json:39]: I don’t think it makes much sense to slow down all the builds for something that is used for debugging.

Json stats should only be used when needed, as this is one of the slowest and resource hungry option.

File: aio/angular.json (hunk 5 above)
Comment (gkalpak) [reply aio/angular.json:42]: I created a new projec with CLI v12.0.0-rc.0 and I  don't get named chinks with `ng build` 😕

Comment (alan-agius4) [reply aio/angular.json:42]: Oops, the named chunks migration has a bug.
https://github.com/angular/angular-cli/pull/20646

Comment (alan-agius4) [reply aio/angular.json]: PS: the new values can be found here: https://next.angular.io/cli/build

**Comment by jessicajaniuk** [comment]
@alan-agius4 This PR had issues with merging and required a rebase. Now there's a conflict with the yarn.lock that needs to be resolved before we can try again.

**Comment by alan-agius4** [comment]
@jessicajaniuk rebase!

**Comment by jessicajaniuk** [comment]
@alan-agius4 This still has issues with merging and I think you'll need to squash. We see errors when addressing individual commits here during the merge process.

```
CONFLICT (content): Merge conflict in aio/angular.json
error: could not apply 99493e2e0b... fixup! build(docs-infra): update angular packages to `12.0.0-rc.0`
```

**Comment by alan-agius4** [comment]
> @alan-agius4 This still has issues with merging and I think you'll need to squash. We see errors when addressing individual commits here during the merge process.
> 
> ```
> CONFLICT (content): Merge conflict in aio/angular.json
> error: could not apply 99493e2e0b... fixup! build(docs-infra): update angular packages to `12.0.0-rc.0`
> ```

done

File: aio/angular.json (hunk 6 above)
Comment (IgorMinar) [reply aio/angular.json:39]: > IIRC we had discussed this in the past and @IgorMinar was keen on keeping the stats.
> I might remember incorrectly, but it would be good to check with @IgorMinar (just in case).

It would be great to keep the stats file as it aids debugging we pack issues.

Comment (IgorMinar) [reply aio/angular.json:39]: > This option is also way more expensive in Webpack 5.

Interesting. I wouldn't have guessed. Maybe we could then turn the stats file generation only for builds that we push to firebase/preview server?

I see the stats file just as important as source maps. You don't always need them but when you do, it's great to have it readily available.

Comment (alan-agius4) [reply aio/angular.json:39]: Indeed it aids debugging Webpack issues, but shouldn’t this be enabled only when needing to debugging?

As mentioned generating stats is very expensive and should only be used when needing them, and not in CI run.

Comment (alan-agius4) [reply aio/angular.json:39]: > Interesting. I wouldn't have guessed. Maybe we could then turn the stats file generation only for builds that we push to firebase/preview server?

Not sure which build that is need to look at bit at the CI setup.

In general since we use a lock file, webpack issues should be reproducible, hence you can clone the repo and run the build with `—stats-json` locally to generate the stats file.
